core/src/test/scala/unit/kafka/server/AbstractApiVersionsRequestTest.scala

```diff
@@ -41,12 +42,12 @@ abstract class AbstractApiVersionsRequestTest(cluster: ClusterInstance) {
     } else {
       cluster.brokerSocketServers().asScala.head
     }
-    IntegrationTestUtils.connectAndReceive[ApiVersionsResponse](request, socket, listenerName)
+    IntegrationTestUtils.connectAndReceive[ApiVersionsResponse](request, socket.boundPort(listenerName))
   }
 
   def sendUnsupportedApiVersionRequest(request: ApiVersionsRequest): ApiVersionsResponse = {
     val overrideHeader = IntegrationTestUtils.nextRequestHeader(ApiKeys.API_VERSIONS, Short.MaxValue)
-    val socket = IntegrationTestUtils.connect(cluster.brokerSocketServers().asScala.head, cluster.clientListener())
+    val socket = IntegrationTestUtils.connect(cluster.boundPorts().asScala.head)
```

Comment:
If `boundPorts` returns `List` instead of `Collection`, the scala conversion could be removed.